src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+	private final CmsItemLookup itemLookup;
+	private final TransformerServiceFactory transformerServiceFactory;
+	private final CmsRepositoryLookup repoLookup;
+	private final Map<String, TransformerService> stylesheets;
+	private final XmlSourceReaderS9api sourceReader;
+	private final TransformerService transformerIdentity;
+	
+	private static final String TRANSFORM_LOCK_COMMENT = "Locked for transform";
+	private static final String TRANSFORM_BASE_PROP_KEY = "abx:TransformBase";
+	private static final String TRANSFORM_NAME_PROP_KEY = "abx:TransformName";
+	private static final Logger logger = LoggerFactory.getLogger(TransformServiceXsl.class);
+
+	@Inject
+	public TransformServiceXsl(
+			CmsCommit commit,
+			CmsItemLookup itemLookup,
+			CmsRepositoryLookup lookupRepo,
+			TransformerServiceFactory transfromerServiceFactory,
+			Map<String, TransformerService> stylesheets,
+			XmlSourceReader sourceReader
+			) {
+		
+		this.commit = commit;
+		this.itemLookup = itemLookup;
+		this.repoLookup = lookupRepo;
+		this.transformerServiceFactory = transfromerServiceFactory;
+		this.transformerIdentity = transfromerServiceFactory.buildTransformerService(new StreamSource(this.getClass().getClassLoader().getResourceAsStream("se/simonsoft/cms/xmlsource/transform/identity.xsl")));
+		this.stylesheets = stylesheets;
+		if (sourceReader instanceof XmlSourceReaderS9api) {
+			this.sourceReader = (XmlSourceReaderS9api) sourceReader;
+		} else {
+			throw new IllegalArgumentException("TransformServiceXsl requires Saxon S9API impl of XmlSourceReader, got: " + sourceReader.getClass());
+		}
+	}
+
+	@Override
+	public void transform(CmsItem item, TransformConfig config) {
+		if (config == null || config.getOptions() == null) {
+			throw new IllegalArgumentException("TransformServiceXsl needs a valid TransformConfig object.");
+		}
+		
+		final CmsItem base = item;
+		final CmsItemId baseItemId = base.getId();
+		final CmsRepository repository = baseItemId.getRepository();
+		final RepoRevision baseRevision = repoLookup.getYoungest(repository);
+		final Boolean overwrite = new Boolean(config.getOptions().getParams().get("overwrite"));
+		
+		if (!config.getOptions().getType().equals("xsl")) {
+			throw new IllegalArgumentException("TransformServiceXsl can only handle xsl transforms but was given: " + config.getOptions().getType());
+		}
+		
+		final String stylesheet = config.getOptions().getParams().get("stylesheet");
+		if (stylesheet == null || stylesheet.trim().isEmpty()) {
+			throw new IllegalArgumentException("Requires a valid stylesheet path or stylesheet name.");
+		}
+		
+		final CmsItemPath outputPath = getOutputPath(baseItemId ,config.getOptions().getParams().get("output"));
+		if (!pathExists(repository, outputPath)) {
+			throw new IllegalArgumentException("Specified output must be an existing folder: " + outputPath.getPath());
+		}
+		
+		final TransformerService transformerService = getTransformerService(baseItemId, stylesheet);
```

Comment:
I don't think this should ever return null.

Reply:
Will never be null. getTransformerService() will throw IAE if stylesheets is uknown.